prime-dotnet-webapi/Services/OrganizationService.cs

```diff
@@ -118,6 +140,52 @@ public async Task<int> CreateOrganizationAsync(int partyId)
             return organization.Id;
         }
 
+        public async Task<Organization> ClaimOrganizationAsync(OrganizationClaimViewModel claimOrganization)
+        {
+            var organization = await GetOrganizationByPecAsync(claimOrganization.PEC);
+            if (organization == null)
+            {
+                return null;
+            }
+
+            var organizationCLaim = new OrganizationClaim
+            {
+                OrganizationId = organization.Id,
+                NewSigningAuthorityId = claimOrganization.PartyId,
+                ProvidedSiteId = claimOrganization.PEC,
+                Details = claimOrganization.ClaimDetail
+            };
+
+            _context.OrganizationClaims.Add(organizationCLaim);
+            await _context.SaveChangesAsync();
+
+            await _businessEventService.CreateOrganizationEventAsync(organization.Id, claimOrganization.PartyId, "Organization Claim Created");
+
+            return organization;
+        }
+
+        public async Task<bool> OrganizationClaimExistsAsync(OrganizationClaimSearchOptions searchOptions)
+        {
+            // return false if searchOptions is invalid
+            if (searchOptions == null || string.IsNullOrEmpty(searchOptions.Pec) && string.IsNullOrEmpty(searchOptions.UserId))
+            {
+                return false;
+            }
+
+            var userId = Guid.Empty;
+            Guid.TryParse(searchOptions.UserId, out userId);
+
+            return await _context.OrganizationClaims
+                .AsNoTracking()
+                .If(!string.IsNullOrEmpty(searchOptions.Pec), q => q
```

Comment:
These `.If()` statements are not needed, you have the guard clause on line 169 ensuring they are not null or empty

Reply:
The guard is to ensure both PEC and userId cannot be empty at the same time, but one can be empty. When PEC is empty and userId is non-empty, we do not want the where clause to search PEC.

Reply:
ah, I see now, thanks